neuralop/layers/attention_kernel_integral.py

```diff
@@ -0,0 +1,212 @@
+import torch
+from torch import nn
+import math
+from torch.nn.init import xavier_uniform_, zeros_
+
+
+class AttentionKernelIntegral(torch.nn.Module):
+    """
+    Kernel integral transform with attention
+    Computes \int_{Omega} k(x, y) * f(y) dy,
+    where:
+          K(x, y) = \sum_{c=1}^d \q_c(x) * \k_c(y)
+          f(y) = v(y)
```

Comment:
Thanks for the comment here. My previous docstring and variable naming might be a bit vague. I added a more detailed description and renamed some variables in the `forward` in the latest commit.

Here is what I added to the docstring and I also renamed variable based on this: 
This module computes $\int_{\Omega} k(x, y) * f(y) dy$,
with:
      $$f(y)=v(y), k(x, y) = \sum_{c=1}^d q_c(x) * k_c(y), \text{where: } q(x) = [q_1(x); ...; q_d(x)], k(y) = [k_1(y); ...;k_d(y)]$$
The learnable weights are $W_q, W_k, W_v \in \mathbb{R}^{d \times d}$.
1. Self-attention:
    There is only a single input function $u$ and its grid $D_x$
    query function: $q(x_i) = a(x_i) W_q$
    key function: $k(x_i) = a(x_i) W_k$
    value function: $v(x_i) = a(x_i) W_v$

2. Cross-attention:
    First input function $u_{qry}$, and its grid $D_x$
    Second input function $u_{src}$, and its grid $D_y$.
    query function: $q(x_i) = u_{qry}(x_i) W_q$
    key function: $k(y_j) = u_{src}(y_j) W_k$
    value function: $v(y_j) = u_{src}(y_j) W_v$

Self-attention can be considered as a special case of cross-attention, where $u = u_{qry} = u_{src}$ and $D_x = D_y$.

Reply:
Great, thanks a lot for the quick update! Two nits:
1. I think the `a` should be a `u` in the first item.
2. We could add the discretization `sum_{i} k(x_i, x_i) f(x_i) w(x_i)` (and analogously for the cross-attention)  where we mention the quadrature weights `w(x_i)`.
2. For self-attention, the `forward` currently seems to support `D_y` to be different from `D_x`.  I think this should raise a `ValueError` since we do not have the matching function values in this case.